inotify.go

```diff
@@ -149,20 +183,31 @@ func (w *Watcher) readEvents() {
 			continue
 		}
 
+		// syscall.Read might have been woken up by Close. If so, we're done.
+		if w.isClosed() {
+			return
+		}
+
 		// If EOF is received
 		if n == 0 {
-			syscall.Close(w.fd)
-			close(w.Events)
-			close(w.Errors)
+			close(w.done)
 			return
 		}
 
 		if n < 0 {
-			w.Errors <- os.NewSyscallError("read", errno)
+			select {
```

Comment:
these two selects make me curious.

we are sending an error or checking if the done channel is closed, whichever happens first?